pkg/maps/lbmap/ipv4.go

```diff
@@ -246,3 +298,172 @@ func NewRevNat4Value(ip net.IP, port uint16) *RevNat4Value {
+func (k *Service4KeyV2) GetAddress() net.IP        { return k.Address.DuplicateIP() }
+func (k *Service4KeyV2) GetPort() uint16           { return k.Port }
+func (k *Service4KeyV2) MapDelete() error          { return k.Map().Delete(k.ToNetwork()) }
+
+func (k *Service4KeyV2) ToNetwork() ServiceKeyV2 {
+	n := *k
+	n.Port = byteorder.HostToNetwork(n.Port).(uint16)
+	return &n
+}
+
+// Service4ValueV2 must match 'struct lb4_service_v2' in "bpf/lib/common.h".
+type Service4ValueV2 struct {
+	Count     uint16 `align:"count"`
+	BackendID uint16 `align:"backend_id"`
+	RevNat    uint16 `align:"rev_nat_index"`
+	Weight    uint16 `align:"weight"`
+}
+
+func NewService4ValueV2(count uint16, backendID uint16, revNat uint16, weight uint16) *Service4ValueV2 {
+	svc := Service4ValueV2{
+		Count:     count,
+		BackendID: backendID,
+		RevNat:    revNat,
+		Weight:    weight,
+	}
+
+	return &svc
+}
+
+func (s *Service4ValueV2) String() string {
+	return fmt.Sprintf("%d (%d)", s.BackendID, s.RevNat)
+}
+
+func (s *Service4ValueV2) GetValuePtr() unsafe.Pointer { return unsafe.Pointer(s) }
+
+func (s *Service4ValueV2) SetCount(count int)      { s.Count = uint16(count) }
+func (s *Service4ValueV2) GetCount() int           { return int(s.Count) }
+func (s *Service4ValueV2) SetRevNat(id int)        { s.RevNat = uint16(id) }
+func (s *Service4ValueV2) GetRevNat() int          { return int(s.RevNat) }
+func (s *Service4ValueV2) SetWeight(weight uint16) { s.Weight = weight }
+func (s *Service4ValueV2) GetWeight() uint16       { return s.Weight }
+func (s *Service4ValueV2) SetBackendID(id uint16)  { s.BackendID = id }
+func (s *Service4ValueV2) GetBackendID() uint16    { return s.BackendID }
+func (s *Service4ValueV2) RevNatKey() RevNatKey    { return &RevNat4Key{s.RevNat} }
+
+func (s *Service4ValueV2) ToNetwork() ServiceValueV2 {
+	n := *s
+	n.RevNat = byteorder.HostToNetwork(n.RevNat).(uint16)
+	n.Weight = byteorder.HostToNetwork(n.Weight).(uint16)
+	return &n
+}
+
+type Backend4Key struct {
+	ID uint16
+}
+
+func NewBackend4Key(id uint16) *Backend4Key {
+	return &Backend4Key{ID: id}
+}
+
+func (k *Backend4Key) String() string            { return fmt.Sprintf("%d", k.ID) }
+func (k *Backend4Key) GetKeyPtr() unsafe.Pointer { return unsafe.Pointer(k) }
+func (k *Backend4Key) NewValue() bpf.MapValue    { return &Backend4Value{} }
+func (k *Backend4Key) Map() *bpf.Map             { return Backend4Map }
+func (k *Backend4Key) SetID(id uint16)           { k.ID = id }
+func (k *Backend4Key) GetID() uint16             { return k.ID }
+
+// Backend4Value must match 'struct lb4_backend' in "bpf/lib/common.h".
+type Backend4Value struct {
+	Address types.IPv4      `align:"address"`
+	Port    uint16          `align:"port"`
+	Proto   u8proto.U8proto `align:"proto"`
+	Pad     uint8
+}
+
+func NewBackend4Value(ip net.IP, port uint16, proto u8proto.U8proto) (*Backend4Value, error) {
+	ip4 := ip.To4()
+	if ip4 == nil {
+		return nil, fmt.Errorf("Not an IPv4 address")
```

Comment:
Removed the redundant check in the `NewBackend{4,6}(...)` methods.

However, I'd prefer to keep the check in `NewBackend{4,6}Value(...)`, because in v1.6 its creation step will change (currently it is derived from `ServiceValue` which IP is checked). So, we could move the validation to a higher level in v1.6.